android/src/main/java/dev/robingenz/capacitorjs/plugins/firebase/auth/FirebaseAuthentication.java

```diff
@@ -128,6 +131,39 @@ public void signInWithYahoo(PluginCall call) {
         oAuthProviderHandler.signIn(call, "yahoo.com");
     }
 
+    public void signInWithCustomToken(PluginCall call) {
+        boolean skipNativeAuth = this.config.getSkipNativeAuth();
+        if (skipNativeAuth) {
+            call.unimplemented(ERROR_CUSTOM_TOKEN_UNIMPLEMENTED);
+            return;
+        }
+
+        String token = call.getString("token");
+        if (token == null) {
+            Log.e(FirebaseAuthenticationPlugin.TAG, "signInWithCustomToken failed: invalid token.");
+            call.reject(ERROR_INVALID_CUSTOM_TOKEN);
+            return;
+        }
```

Comment:
```suggestion
        String token = call.getString("token", "");
```

I would keep the code as simple as possible at this point. 
Extensive error handling will be added soon.